### PR TITLE
Add Safari versions for MutationRecord API

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR adds real values for Safari (Desktop and iOS/iPadOS) for the `MutationRecord` API, based upon results from the [mdn-bcd-collector](https://mdn-bcd-collector.appspot.com) project (v1.1.6).  Results are manually confirmed for accuracy.

Tests Used: https://mdn-bcd-collector.appspot.com/tests/api/MutationRecord

### CODE REVIEW

File: api/MutationRecord.json [hunk 1]
```diff
@@ -29,10 +29,10 @@
             "version_added": true
           },
           "safari": {
-            "version_added": true
+            "version_added": "7"
```
Comment (foolip) [inline api/MutationRecord.json:32]: Going by https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#Browser_compatibility it seems more likely this was introduced in Safari 6, but perhaps the interface wasn't exposed then?

Comment (vinyldarkscratch) [reply api/MutationRecord.json:32]: It looks like `MutationObserver` was supported in Safari 6 behind the `WebKit` prefix, so it's probably the same case here.  Digging deeper into this.

Comment (vinyldarkscratch) [reply api/MutationRecord.json:32]: Hmm...nope, not a prefix.  It _is_ supported in Safari 6.1, though.  I wonder why "6.2" keeps getting translated into "7"...

Comment (foolip) [reply api/MutationRecord.json]: BCD doesn't have 6.2 as a version. The version matching logic should map it to 6.1, but maybe there's a bug?

**Review by foolip — APPROVED** [review]

**Pull request merged**